sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/MemorySinkSuite.scala

```diff
@@ -220,11 +220,11 @@ class MemorySinkSuite extends StreamTest with BeforeAndAfter {
 
     sink.addBatch(0, 1 to 3)
     plan.invalidateStatsCache()
-    assert(plan.stats.sizeInBytes === 12)
+    assert(plan.stats.sizeInBytes === 36)
 
     sink.addBatch(1, 4 to 6)
     plan.invalidateStatsCache()
-    assert(plan.stats.sizeInBytes === 24)
+    assert(plan.stats.sizeInBytes === 72)
```

Comment:
`MemorySinkV2 ` is mainly for testing. I think the stats changes will not impact anything, right? @tdas @jose-torres

Reply:
It shouldn't impact anything, but abstractly it seems strange that this unification would cause the stats to change? What are we doing differently to cause this, and how confident are we this won't happen to production sinks?

Reply:
It seems we forgot to count the row object overhead (8 bytes) before in memory stream.

Reply:
SGTM then